xchange-stream-kraken/src/main/java/info/bitrich/xchangestream/kraken/KrakenStreamingService.java

```diff
@@ -163,6 +163,19 @@ public String getSubscribeMessage(String channelName, Object... args) throws IOE
       }
       subscriptionRequestMap.put(reqID, channelName);
 
+      Observable.timer(2, TimeUnit.SECONDS)
+          .takeWhile(t -> isSocketOpen())
+          .subscribe(
+              t -> {
+                String chName = subscriptionRequestMap.get(reqID);
+                if (chName != null && chName.equals(channelName)) {
+                  LOG.error("Unable to subscribe channel {}", channelName);
```

Comment:
We should be a little more explicit here, I think, just to be helpful, e.g. "Failed to receive subscription confirmation from Kraken within 2 seconds". And in the channel error.

Reply:
You are right. I improved the error message.